### PR TITLE
[PAN-2444]  Add CLI flag for setting WorldStateDownloader task cache size

### PR DESCRIPTION
**Pull request by matkt**
## PR description

Add  `--fast-sync-task-cache-size` command

### CODE REVIEW

File: pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java [hunk 1]
```diff
@@ -291,6 +291,13 @@ void setBannedNodeIds(final List<String> values) {
           "Minimum number of peers required before starting fast sync. (default: ${DEFAULT-VALUE})")
   private final Integer fastSyncMinPeerCount = FAST_SYNC_MIN_PEER_COUNT;
 
+  @Option(
+      names = {"--fast-sync-task-cache-size"},
+      paramLabel = MANDATORY_INTEGER_FORMAT_HELP,
+      description =
+          "The size of the cache containing the tasks to be operated. (default: ${DEFAULT-VALUE})")
+  private final Integer taskCollectionCacheSize = DEFAULT_TASK_CACHE_SIZE;
```
Comment (mbaxter) [inline pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java]: Can we move this option into `SynchronizerOptions` and use the "experimental" flag style (beginning with `--X`) similar to: https://github.com/PegaSysEng/pantheon/blob/12d19a152d238ea5629e20997963865aa4784ee2/pantheon/src/main/java/tech/pegasys/pantheon/cli/options/SynchronizerOptions.java#L56-L65

File: pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java [hunk 2]
```diff
@@ -1343,6 +1343,28 @@ public void parsesInvalidFastSyncMinPeersOptionWrongFormatShouldFail() {
         .contains("Invalid value for option '--fast-sync-min-peers': 'ten' is not an int");
   }
 
+  @Test
+  public void parsesValidFastSyncTaskCacheSizeOption() {
```
Comment (mbaxter) [inline pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java]: If we move this flag to the experimental style in `SychronizerOptions` as suggested above, you should be able to cut these tests and just add a call to customize the cache size options here: https://github.com/PegaSysEng/pantheon/blob/12d19a152d238ea5629e20997963865aa4784ee2/pantheon/src/test/java/tech/pegasys/pantheon/cli/options/SynchronizerOptionsTest.java#L31-L61

**Comment by matkt** [comment]
> Thanks for the submission @matkt ! This looks really good - just left a few minor comments to address before approving.

Thanks you for your review. I made the changes. Do not hesitate to review

File: pantheon/src/test/java/tech/pegasys/pantheon/cli/options/SynchronizerOptionsTest.java [hunk 3]
```diff
@@ -41,6 +41,7 @@
             SynchronizerConfiguration.DEFAULT_WORLD_STATE_MAX_REQUESTS_WITHOUT_PROGRESS * 2)
         .worldStateMinMillisBeforeStalling(
             SynchronizerConfiguration.DEFAULT_WORLD_STATE_MIN_MILLIS_BEFORE_STALLING * 2)
+        .worldStateTaskCacheSize(SynchronizerConfiguration.DEFAULT_WORLD_STATE_TASK_CACHE_SIZE)
```
Comment (mbaxter) [inline pantheon/src/test/java/tech/pegasys/pantheon/cli/options/SynchronizerOptionsTest.java]:
```suggestion
        .worldStateTaskCacheSize(SynchronizerConfiguration.DEFAULT_WORLD_STATE_TASK_CACHE_SIZE + 1)
```

File: pantheon/src/main/java/tech/pegasys/pantheon/cli/options/SynchronizerOptions.java [hunk 4]
```diff
@@ -196,6 +198,16 @@ public void parseBlockPropagationRange(final String arg) {
   private long worldStateMinMillisBeforeStalling =
       SynchronizerConfiguration.DEFAULT_WORLD_STATE_MIN_MILLIS_BEFORE_STALLING;
 
+  @CommandLine.Option(
+      names = WORLD_STATE_TASK_CACHE_SIZE_FLAG,
+      hidden = true,
+      defaultValue = "1000000",
+      paramLabel = "<INTEGER>",
+      description =
+          "The size of the cache containing the tasks to be operated. (default: ${DEFAULT-VALUE})")
```
Comment (mbaxter) [inline pantheon/src/main/java/tech/pegasys/pantheon/cli/options/SynchronizerOptions.java]:
```suggestion
          "The max number of pending node data requests cached in-memory during fast sync world state download. (default: ${DEFAULT-VALUE})")
```

**Pull request merged**